app/javascript/packages/document-capture/components/in-person-location-step-post-office-search-step.tsx

```diff
@@ -0,0 +1,214 @@
+  const [foundAddress, setFoundAddress] = useState({} as LocationQuery);
+  const [inProgress, setInProgress] = useState(false);
+  const [autoSubmit, setAutoSubmit] = useState(false);
+  const [isLoadingComplete, setIsLoadingComplete] = useState(false);
+  const { setSubmitEventMetadata } = useContext(AnalyticsContext);
+  const { arcgisSearchEnabled } = useContext(InPersonContext);
+
+  // ref allows us to avoid a memory leak
+  const mountedRef = useRef(false);
+
+  useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  // useCallBack here prevents unnecessary rerenders due to changing function identity
+  const handleLocationSelect = useCallback(
+    async (e: any, id: number) => {
+      const selectedLocation = locationData[id];
+      const { name: selectedLocationName } = selectedLocation;
+      setSubmitEventMetadata({ selected_location: selectedLocationName });
+      onChange({ selectedLocationName });
+      if (autoSubmit) {
+        return;
+      }
+      // prevent navigation from continuing
+      e.preventDefault();
+      if (inProgress) {
+        return;
+      }
+      const selected = prepToSend(selectedLocation);
+      setInProgress(true);
+      await request(LOCATIONS_URL, {
+        json: selected,
+        method: 'PUT',
+      })
+        .then(() => {
+          if (!mountedRef.current) {
+            return;
+          }
+          setAutoSubmit(true);
+          setImmediate(() => {
+            // continue with navigation
+            e.target.click();
+            // allow process to be re-triggered in case submission did not work as expected
+            setAutoSubmit(false);
+          });
+        })
+        .finally(() => {
+          if (!mountedRef.current) {
+            return;
+          }
+          setInProgress(false);
+        });
+    },
+    [locationData, inProgress],
+  );
+
+  const handleFoundAddress = useCallback((address) => {
+    setFoundAddress({
+      streetAddress: address.street_address,
+      city: address.city,
+      state: address.state,
+      zipCode: address.zip_code,
+    });
+  }, []);
+
+  useEffect(() => {
+    let didCancel = false;
+    (async () => {
+      try {
+        const fetchedLocations = await getUspsLocations(prepToSend(foundAddress));
+
+        if (!didCancel) {
+          const formattedLocations = formatLocation(fetchedLocations);
+          setLocationData(formattedLocations);
+        }
+      } finally {
+        if (!didCancel) {
+          setIsLoadingComplete(true);
+        }
+      }
+    })();
+    return () => {
+      didCancel = true;
+    };
+  }, [foundAddress]);
+
+  let locationsContent: React.ReactNode;
+  if (!isLoadingComplete) {
+    locationsContent = <SpinnerDots />;
+  } else if (locationData.length < 1) {
+    locationsContent = <h4>{t('in_person_proofing.body.location.none_found')}</h4>;
+  } else {
+    locationsContent = (
+      <LocationCollection>
```

Comment:
Will this existing content continue to be relevant in the redesigned step? Should we leave it out?